packages/react/src/primitives/Menu/Menu.tsx

```diff
@@ -40,9 +40,16 @@ const MenuPrimitive: Primitive<MenuProps, 'div'> = (
   ref
 ) => {
   const icons = useIcons('menu');
+
+  const shouldBeDisabled =
+    // This defaults to false if isDisabled and disabled are undefined
+    // @ts-ignore: Property 'props' does not exist on type 'string'.
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
```

Comment:
Yeah def open to suggestions here. As is we can't infer the exact type of `trigger` ahead of time so not sure how to get around this. 

I honestly think we should be enforcing a type for `trigger` because if you pass a non JSX element as a child of DropdownMenuTrigger, nothing shows up. I can't find the radix documentation on this but it seems like they don't support things that aren't JSX elements in the version we're using. But I think that would be a breaking change so not sure if that's something we can do.

Reply:
Take a look at the `React.cloneElement` API